### ISSUE TITLE
FATAL ERROR: JS Allocation failed - process out of memory

### ISSUE THREAD

**Issue by irvinebroque** [awesomebox/awesomebox#9]
Memory leak? Issue dealing with images? I'm starting to see this issue as I'm using larger media files with Awesomebox.

**Comment by mattinsler** [comment]
oof... not sure, but definitely not a good thing! can you give me an example of a file? should i just try any large image?

**Comment by irvinebroque** [comment]
I can't tell if it's specific to a file, because things we're working fine, and then I hit a LESS compile error, which usually is fine (change, save, works again), but this time it terminated the process and generated this error.

Looking at active processes, it's CPU bound - my laptop fan is going nuts:

![screen shot 2013-08-18 at 2 22 50 pm](https://f.cloud.github.com/assets/1221592/983259/6bf73402-084c-11e3-8df3-f9f4b4b2a2f9.png)

**Comment by mattinsler** [comment]
try running it again, but with --no-watch... i'm wondering if the CPU is because it's file watching

**Comment by irvinebroque** [comment]
Good call - that is indeed what's making the CPU spike. Is the file-watching just livereload or something else?

**Comment by mattinsler** [comment]
Yes, it's only for livereload. However livereload itself is not the problem. It's the file watching stuff inside of node.js that's the main problem. If you were using brackets to edit files and brackets would tell you which files had changed and when, then awesomebox could use that rather than file watching to trigger livereload. That's a potential integration for different text editors, all of whom do their own file watching in way less intense ways.
